daemon/config.go

```diff
@@ -121,6 +125,10 @@ type CommonConfig struct {
 	// may take place at a time for each push.
 	MaxConcurrentUploads *int `json:"max-concurrent-uploads,omitempty"`
 
+	// ShutdownTimeout is the timeout value (in seconds) the daemon will wait for the container
+	// to stop when daemon is being shutdown
+	ShutdownTimeout int `json:"shutdown-timeout,omitempty"`
```

Comment:
I would expect this to be handled on the client, not the API (even though I'm sure we probably do this elsewhere...)